### PR TITLE
added type-annotations and fixed the code

### PR DESCRIPTION
**Pull request by SegFault03**
Did some refactoring.
Added type annotations.
Code works now. 

See results: https://freeimage.host/i/juVoes

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    aWeight = 0.5
+
+    # get the reference to the webcam
+    camera = cv2.VideoCapture(0)
+
+    # region of interest (ROI) coordinates
+    top, right, bottom, left = 10, 350, 225, 590
+
+    # initialize num of frames
+    num_frames = 0
```
Comment (soumyadeepkun) [inline src/main.py]: don't use camel and snake cases for fixed-value variables.
Ex: change `aWeight` to `a_weight`

File: src/main.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import cv2
+import imutils
+import numpy as np
+
+# global variables
+BG = None
+
+def run_avg(image: np.ndarray, aWeight: float)->None:
```
Comment (soumyadeepkun) [inline src/main.py]: Leave a whitespace before and after the "arrow" `(->)` when specifying type annotations. 
Ex: 
```python
def run_avg(image: np.ndarray, a_weight: float) -> None:
```

File: src/main.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    if len(cnts) == 0:
+        return
+    else:
+        # based on contour area, get the maximum contour which is the hand
+        segmented = max(cnts, key=cv2.contourArea)
+        return (thresholded, segmented)
+
+#-----------------
+# MAIN FUNCTION
+#-----------------
```
Comment (soumyadeepkun) [inline src/main.py]: This is neither necessary, nor the correct interpretation of the purpose of the `__name__` variable. Consider removing these lines.

**Review by soumyadeepkun — CHANGES_REQUESTED** [review]

**Comment by soumyadeepkun** [comment]
Good work on fixing the code. Just make the suggested changes and all would be good to go.

File: src/main.py (hunk 3 above)
Comment (SegFault03) [reply src/main.py]: literally a comment, but ok.

**Comment by SegFault03** [comment]
All done. 
Pre-commit hook ran fine.
Updated requirements.txt

**Review by soumyadeepkun — APPROVED** [review]

**Comment by soumyadeepkun** [comment]
LGTM.

**Pull request merged**